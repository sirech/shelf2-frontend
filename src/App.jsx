import React, { Component } from 'react'
import { BrowserRouter as Router, Route } from 'react-router-dom'

import { Container } from 'reactstrap'

import Header from './molecules/header'

class App extends Component {
  render () {
    return (
      <Router>
        <div>
          <Header />
          <Container fluid>
            <Route exact path='/' component={Home} />
          </Container>
        </div>
      </Router>
    )
  }
}

const Home = () => (
  <p className='App-intro'>
    To get started, edit <code>src/App.js</code> and save to reload.
  </p>
)

export default App
