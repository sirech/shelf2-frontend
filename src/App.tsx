import React, { Component } from 'react'
import {
  BrowserRouter as Router,
  Navigate,
  Route,
  Routes,
} from 'react-router-dom'

import Header from 'header'
import BooksArea from 'books_area'

import SearchList from 'search_list'
import NewBook from 'new_book'

import Callback from 'callback'

const year = new Date().getFullYear()

class App extends Component {
  render() {
    return (
      <Router>
        <div>
          <Header />
          <Content />
        </div>
      </Router>
    )
  }
}

const Content = () => (
  <Routes>
    <Route path="/" element={<Navigate replace to="/books" />} />
    <Route path="/books" element={<Navigate to={`/books/${year}`} />} />
    <Route path="/books/search" element={<Navigate replace to="/books" />} />
    <Route path="/books/search/:keyword" element={<SearchList />} />
    <Route path="/books/new" element={<NewBook />} />
    <Route path="/books/:year" element={<BooksArea />} />
    <Route path="/callback" element={<Callback />} />
  </Routes>
)

export default App
