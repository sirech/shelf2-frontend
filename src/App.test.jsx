import React from 'react'
import { fullRender } from 'test'

import App from './App'

jest.mock('rest/fetch')

describe('components', () => {
  describe('App', () => {
    it('renders correctly', () => {
      const { component } = fullRender(<App />)
      expect(component.toJSON()).toMatchSnapshot()
    })
  })
})
