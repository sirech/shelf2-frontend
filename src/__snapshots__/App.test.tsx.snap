// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components App renders correctly 1`] = `
<div>
  <div />
  <header
    className="mb-4"
  >
    <nav
      className="navbar navbar-expand-sm navbar-dark bg-dark"
    >
      <a
        className="active navbar-brand"
        href="/books"
        onClick={[Function]}
        style={Object {}}
      >
        Shelf
      </a>
      <button
        aria-label="Toggle navigation"
        className="navbar-toggler"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        style={Object {}}
      >
        <ul
          className="ml-auto navbar-nav"
        >
          <form
            className="form-inline"
          >
            <input
              className="form-control"
              onChange={[Function]}
              type="text"
            />
          </form>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
          <li
            className="nav-item"
          >
            <div
              className="nav-link"
              onClick={[Function]}
              role="link"
            >
              Logout
            </div>
          </li>
        </ul>
      </div>
    </nav>
  </header>
  <div
    className="grid"
  >
    <main>
      <span
        className="createButton"
      >
        <button
          aria-label={null}
          className="btn btn-success"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-plus fa-w-14 "
            data-icon="plus"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          Add Book
        </button>
      </span>
      <div />
    </main>
    <aside
      className="list-group"
    />
  </div>
</div>
`;
